cockatrice/src/dialogs/dlg_move_top_cards_until.cpp

```diff
@@ -0,0 +1,66 @@
+#include "dlg_move_top_cards_until.h"
+
+#include "../game/filters/filter_string.h"
+#include "trice_limits.h"
+
+#include <QDialogButtonBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMessageBox>
+#include <QSpinBox>
+#include <QString>
+#include <QVBoxLayout>
+#include <QWidget>
+
+DlgMoveTopCardsUntil::DlgMoveTopCardsUntil(QWidget *parent, QString _expr, uint _numberOfHits) : QDialog(parent)
+{
+    exprLabel = new QLabel(tr("Card name (or search expressions):"));
+
+    exprEdit = new QLineEdit(this);
+    exprEdit->setFocus();
+    exprEdit->setText(_expr);
+    exprLabel->setBuddy(exprEdit);
+
+    numberOfHitsLabel = new QLabel(tr("Number of hits:"));
+    numberOfHitsEdit = new QSpinBox(this);
+    numberOfHitsEdit->setRange(1, 99);
+    numberOfHitsEdit->setValue(_numberOfHits);
+    numberOfHitsLabel->setBuddy(numberOfHitsEdit);
+
+    auto *grid = new QGridLayout;
+    grid->addWidget(numberOfHitsLabel, 0, 0);
+    grid->addWidget(numberOfHitsEdit, 0, 1);
+
+    buttonBox = new QDialogButtonBox(QDialogButtonBox::Ok | QDialogButtonBox::Cancel);
+    connect(buttonBox, &QDialogButtonBox::accepted, this, &DlgMoveTopCardsUntil::validateAndAccept);
+    connect(buttonBox, &QDialogButtonBox::rejected, this, &QDialog::reject);
+
+    auto *mainLayout = new QVBoxLayout;
+    mainLayout->addWidget(exprLabel);
+    mainLayout->addWidget(exprEdit);
+    mainLayout->addItem(grid);
+    mainLayout->addWidget(buttonBox);
+
+    setLayout(mainLayout);
+    setWindowTitle(tr("Put top cards on stack until..."));
+}
+
+void DlgMoveTopCardsUntil::validateAndAccept()
+{
+    auto movingCardsUntilFilter = FilterString(exprEdit->text());
+    if (movingCardsUntilFilter.valid()) {
+        accept();
+    } else {
+        QMessageBox::warning(this, tr("Invalid filter"), movingCardsUntilFilter.error(), QMessageBox::Ok);
```

Comment:
I know this currently doesn't work, but I'd like to instead have a message in the existing dialog instead of another dialog

Reply:
Can you give me an example of what that should look like?

Reply:
a label with red text is most common I think, cockatrice doesn't do this anywhere currently afaik but it's extremely common in other apps and websites, feel free to ignore this for now and we can get back to it later

Reply:
sure, let's get the validation to actually work first so we can actually see the validation error